definitions/npm/react-i18next_v1.10.x/flow_v0.36.x-/react-i18next_v1.10.x.js

```diff
@@ -0,0 +1,35 @@
+import type { TFunction } from 'i18next'
+
+declare module 'react-i18next' {
+  declare type TFunction = TFunction;
+  declare type Locales = string | Array<string>;
+
+  declare type StatelessComponent<P> = (props: P) => ?React$Element<any>;
+
+  declare type Comp<P> = StatelessComponent<P> | Class<React$Component<*, P, *>>;
+
+  declare type Translator<OP, P> = {
+    (component: StatelessComponent<P>): Class<React$Component<void, OP, void>>;
+    <Def, St>(component: Class<React$Component<Def, P, St>>): Class<React$Component<Def, OP, St>>;
```

Comment:
By intuition, I think that (1) they are not equivalent... by putting the generic definitions in front of the function paranthesis, it will always redo the check for each function call, so you cannot lock down the type to a specific type in forefront (unless you do some subtyping stuff).

For (2), yes <B> would be of the same type as bar (Bar<X, Y> -> A, B are only in the context of Bar, so I call them X, Y)... so <Bar<X, Y>>(bar: Bar<A, Bar<X, Y>>): Bar<A, Bar<X,Y>>

It's hard to have discussions on generics without sounding very confusing :-)